### PR TITLE
Redesign cbar and panes to allow more menu options

### PR DESCRIPTION
**Pull request by hapurubokka**

### CODE REVIEW

File: src/app/views/discover/communities/communities.ts [hunk 1]
```diff
@@ -62,7 +62,7 @@ export default class RouteDiscoverCommunities extends BaseRouteComponent {
 	}
 
 	get showCreateCommunity() {
-		return this.app.user && !!this.app.user.can_create_communities && !this.hasMore;
+		return (!this.app.user || !!this.app.user.can_create_communities) && !this.hasMore;
```
Comment (nilllzz) [inline src/app/views/discover/communities/communities.ts:65]: Hm, this made it kinda hard to read.
I suggest just doing an early-out user check and then hasMore and can_create_communities.
Something like
```
if (!this.app.user) {
    return false;
}

return !!this.app.user.can_create_communities && !this.hasMore;
```

Comment (hapurubokka) [reply src/app/views/discover/communities/communities.ts:65]: The reason this was changed is that we will be showing the "Add Community" widget for non-users as well, using the `AppAuthRequired` directive. We're checking if we either have no user, or if the user is able to create more communities.

File: src/_common/sidebar/sidebar.store.ts [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+	/** The most recently set context pane */
+	get activeContextPane() {
+		if (this._contextPanes.length > 0) {
+			return this._contextPanes[this._contextPanes.length - 1];
+		}
+
+		return null;
+	}
+
+	/**
```
[the remark this answers is not in the document]
Comment (hapurubokka) [reply src/_common/sidebar/sidebar.store.ts]: revised: Moved the contextual hover information to the `SidebarStore` import.

File: src/app/components/shell/sidebar/context/channels/channels.ts [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+import AppNavEdit from '../../../../../views/communities/view/_nav/edit/edit.vue';
+
+@Component({
+	components: {
+		AppCommunitiesViewCard,
+		AppNavChannels,
+		AppNavEdit,
+	},
+})
+export default class AppShellSidebarContextChannels extends Vue {
```
[the remark this answers is not in the document]
Comment (hapurubokka) [reply src/app/components/shell/sidebar/context/channels/channels.ts]: revised: Moved this file before review - file is now named `AppCommunitiesViewContext` under `src\app\views\communities\view\_context\context.ts`

File: src/app/views/communities/view/view.ts [hunk 4]
```diff
@@ -77,16 +77,30 @@ export default class RouteCommunitiesView extends BaseRouteComponent {
...
+	@SidebarState activeContextPane!: SidebarStore['activeContextPane'];
+	@SidebarMutation addContextPane!: SidebarStore['addContextPane'];
+	@SidebarMutation removeContextPane!: SidebarStore['removeContextPane'];
+	@Action showContextPane!: Store['showContextPane'];
+
 	readonly Environment = Environment;
 	readonly Screen = Screen;
+	readonly sidebarComponent = AppCommunitiesViewContext;
+
+	contextPane: null | ContextPane = null;
```
[the remark this answers is not in the document]
Comment (hapurubokka) [reply src/app/views/communities/view/view.ts]: I'm not sure if we've reached a consensus on this, but I changed it anyway.

File: src/app/components/community/channel/card/card.ts [hunk 5]
```diff
@@ -1,20 +1,26 @@
...
 import {
 	Community,
 	CommunityPresetChannelType,
 } from '../../../../../_common/community/community.model';
 import AppMediaItemBackdrop from '../../../../../_common/media-item/backdrop/backdrop.vue';
 import { MediaItem } from '../../../../../_common/media-item/media-item-model';
+import { AppObserveDimensions } from '../../../../../_common/observe-dimensions/observe-dimensions.directive';
 import { AppTooltip } from '../../../../../_common/tooltip/tooltip-directive';
 
+// Sync up with the './variables' file.
```
[the remark this answers is not in the document]
Comment (hapurubokka) [reply src/app/components/community/channel/card/card.ts:13]: revised: Added the note in the `variables.styl` file - changed the `ts` file to the same value, and modified the `vue` file and styling to get within a reasonable margin of error for the aspect ratio.

File: src/app/views/communities/view/_nav/edit/edit.vue [hunk 6]
```diff
@@ -81,20 +101,14 @@
 				</router-link>
 			</app-community-perms>
 		</ul>
-	</component>
+	</nav>
 </template>
 
+<script lang="ts" src="./edit"></script>
```
[the remark this answers is not in the document]
Comment (hapurubokka) [reply src/app/views/communities/view/_nav/edit/edit.vue]: reformatted all `vue` files - should be all good now!

File: src/app/views/discover/communities/communities.ts (hunk 1 above)
Comment (nilllzz) [reply src/app/views/discover/communities/communities.ts:65]: Oh I see, that's fine then.

**Review by nilllzz — APPROVED** [review]

**Pull request merged**